core/src/main/scala/org/http4s/multipart/Part.scala

```diff
@@ -29,10 +34,41 @@ object Part {
   def empty[F[_]]: Part[F] =
     Part(Headers.empty, EmptyBody)
 
-  def formData[F[_]: Sync](name: String, value: String, headers: Header*): Part[F] =
+  def formData[F[_]: Sync](name: String, value: String, headers: Header*): Part[F] = {
+    val body = Stream.emit(value).through(utf8Encode)
+    val bodyLength = body.as(1).compile.fold(0L)(_ + _)
     Part(
       Headers(`Content-Disposition`("form-data", Map("name" -> name)) :: headers.toList),
-      Stream.emit(value).through(utf8Encode))
+      Stream.emit(value).through(utf8Encode),
+      Some(bodyLength))
+  }
+
+  def fileDataKnownContentLength[F[_]: Sync: ContextShift](
+      name: String,
+      path: Path,
+      blockingEC: ExecutionContext,
+      headers: Header*): F[Part[F]] =
+    fileDataKnownContentLength(
+      name,
+      path.toFile.getName,
+      readAll[F](path, blockingEC, ChunkSize),
+      headers: _*)
+
+  def fileDataKnownContentLength[F[_]: Sync](
```

Comment:
I think these need a warning that they will hold all of `entityBody`/`path` (respectively) in memory; the "default" assumption is usually the opposite.

Reply:
Yes, 100% agree